From 8e6bbc0c94cd203b23c13c344f46d47cfd19f31d Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Sun, 31 Mar 2024 00:24:11 +0000
Subject: regress: Redirect conch stdin from /dev/zero

This is more convenient than requiring a controlling terminal.

Forwarded: https://bugzilla.mindrot.org/show_bug.cgi?id=3676
Last-Update: 2024-03-31

Patch-Name: regress-conch-dev-zero.patch
---
 regress/conch-ciphers.sh | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/regress/conch-ciphers.sh b/regress/conch-ciphers.sh
index 26b606d65..22168570c 100644
--- a/regress/conch-ciphers.sh
+++ b/regress/conch-ciphers.sh
@@ -7,10 +7,6 @@ if test "x$REGRESS_INTEROP_CONCH" != "xyes" ; then
 	skip "conch interop tests not enabled"
 fi
 
-if ! [ -t 0 ]; then
-	skip "conch interop tests requires a controlling terminal"
-fi
-
 start_sshd
 
 for c in aes256-ctr aes256-cbc aes192-ctr aes192-cbc aes128-ctr aes128-cbc \
@@ -21,7 +17,7 @@ for c in aes256-ctr aes256-cbc aes192-ctr aes192-cbc aes128-ctr aes128-cbc \
 	# in conch
 	${CONCH} --identity $OBJ/ssh-ed25519 --port $PORT --user $USER -e none \
 	    --known-hosts $OBJ/known_hosts --notty --noagent --nox11 -n \
-	    127.0.0.1 "cat ${DATA}" 2>/dev/null | cat > ${COPY}
+	    127.0.0.1 "cat ${DATA}" </dev/zero 2>/dev/null | cat > ${COPY}
 	if [ $? -ne 0 ]; then
 		fail "ssh cat $DATA failed"
 	fi
